cpp/ps0005/ps0005_sol.cpp

```diff
@@ -4,50 +4,76 @@ using namespace std;
-    char o ;
-    int r ;
-    cout<<"Enter the two operands:\n" ;
-    cin>> a>> b ;
-    cout<< "Enter an operator from these (+ - * / %): \n" ;
-    cin >> o ;
-    switch (o)
+    // prompting user for 2 operands and an operator 1.5/2
+    double num1, num2;
+    char op;
+    double res;
+    cout << "Enter the two operands:\n";
+    cin >> num1 >> num2;
+    cout << "Enter an operator from these (+ - * / %): \n";
+    cin >> op;
+
+    // checking user inputs and processing 2.5/4
+    // printing the result if all data is valid 1/1
+    switch (op)
     {
-case '+':
-    r = a+b ;
-    cout <<"("<<a<<") "<<"+"<<" ("<<b<<") "<<"= "<< r ;
-    break ;
-case '-' :
-    r = a-b ;
-    cout <<"("<<a<<") "<<"-"<<" ("<<b<<") "<<"= "<< r ;
-    break ;
-case '*':
-    r = a *b ;
-    cout <<"("<<a<<") "<<"*"<<" ("<<b<<") "<<"= "<< r ;
-    break ;
-case '%':
-    r = a % b ;
-    cout <<"("<<a<<") "<<"%"<<" ("<<b<<") "<<"= "<< r ;
-    break ;
-case '/':
-        if (a == 0 && b == 0)
+        case '+':
         {
-            cout <<  "Infinite number of solutions: (0 / 0) is Undefined Quantity.\n" ;
+            res = num1 + num2;
+            cout << "(" << num1 << ") " << "+" << " (" << num2 << ") " << "= " << res;
         }
-        else if ( b == 0)
+        break;
+        
+        case '-':
         {
-            cout << "Can't divide by zero \n" ;
+            res = num1 - num2;
+            cout << "(" << num1 << ") " << "-" << " (" << num2 << ") " << "= " << res;
         }
-        else
+        break;
+
+        case '*':
         {
-            r = a / b ;
-            cout <<"("<<a<<") "<<"/"<<" ("<<b<<") "<<"= "<< r ;
+            res = num1 * num2;
+            cout << "(" << num1 << ") " << "*" << " (" << num2 << ") " << "= " << res;
         }
-        break ;
-    default :
-        cout << " ERROR.\n" ;
+        break;
 
+        case '%':
+        {
+            if ((num1 != (int) num1) || (num2 != (int) num2))
+            {
+                cout << "cannot execute modulo operation for floating point numbers.";
+            }
+            else
+            {
+                res = ((int) num1) % ((int) num2);
+                cout << "(" << num1 << ") " << "%" << " (" << num2 << ") " << "= " << res;
+            } // 0/1
         }
-
-         return 0;
+        break;
+        
+        case '/':
+        {
+            if (num1 == 0 && num2 == 0)
+            {
+                cout << "Infinite number of solutions: (0 / 0) is Undefined Quantity.\n";
+            }
+            else if (num2 == 0)
+            {
+                cout << "Can't divide by zero \n";
+            }
+            else
+            {
+                res = num1 / num2;
+                cout << "(" << num1 << ") " << "/" << " (" << num2 << ") " << "= " << res;
+            } // 1/1
         }
+        break;
+        
+        default:
+            cout << " ERROR.\n"; // 1/1
+    } // 0.5/1
+
+    return 0;
+}
+// good work 👍 7/10

```

Comment:
- [x] program is in the correct path
- [x] no syntax errors while compilation
- [ ] formatted code to be clear and readable
- [ ] no bugs or warnings while execution
- [ ] commented some phrases for blocks of code
- [ ] main returns non-zero/zero number according to the case
+1 bonus mark for your effort ⭐

you can check the [problem solution](https://onlinegdb.com/9tQHZBXgL) for more